tests/manager/manager-ctor-auth_mechanism-error-001.phpt

```diff
@@ -33,6 +33,14 @@ echo throws(function() {
     new MongoDB\Driver\Manager('mongodb://localhost:27017/', ['password' => 'password', 'authMechanism' => 'MONGODB-X509']);
 }, "MongoDB\Driver\Exception\InvalidArgumentException"), "\n";
 
+echo throws(function() {
+    new MongoDB\Driver\Manager('mongodb://localhost:27017/?authSource=foo');
+}, "MongoDB\Driver\Exception\InvalidArgumentException"), "\n";
```

Comment:
I don't understand how this passes. The validation logic was added in `php_phongo_uri_finalize_auth`, but that is only called from `php_phongo_apply_options_to_uri`, which itself returns early if there are no options to process. That suggests that this case would depend on [CDRIVER-3618](https://jira.mongodb.org/browse/CDRIVER-3618) where libmongoc would apply the validation to the final `mongoc_uri_t`.

Do let me know if I'm missing something.

Reply:
The exception is thrown in `php_phongo_make_uri` due to not receiving a `mongoc_uri_t` from `mongoc_uri_new_with_error`. The "Default authentication mechanism requires username" messages originates [in libmongoc](https://github.com/mongodb/mongo-c-driver/blob/2307b56f4f50d6d65409af2f2cfb7ab51bff1124/src/libmongoc/src/mongoc/mongoc-uri.c#L1392..L1397). This particular code was added fairly recently (September 2019), but was never copied over to PHPC.